src/networks/devnet/mod.rs

```diff
@@ -6,132 +6,134 @@ use once_cell::sync::Lazy;
             bundle: None,
         },
         HeightInfo {
             height: Height::Orange,
-            epoch: -11,
+            epoch: get_upgrade_height("FOREST_ORANGE_HEIGHT", -11),
             bundle: None,
         },
         HeightInfo {
             height: Height::Trust,
-            epoch: -13,
+            epoch: get_upgrade_height("FOREST_ACTORSV3_HEIGHT", -13),
             bundle: None,
         },
         HeightInfo {
             height: Height::Norwegian,
-            epoch: -14,
+            epoch: get_upgrade_height("FOREST_NORWEGIAN_HEIGHT", -14),
             bundle: None,
         },
         HeightInfo {
             height: Height::Turbo,
-            epoch: -15,
+            epoch: get_upgrade_height("FOREST_ACTORSV4_HEIGHT", -15),
             bundle: None,
         },
         HeightInfo {
             height: Height::Hyperdrive,
-            epoch: -16,
+            epoch: get_upgrade_height("FOREST_HYPERDRIVE_HEIGHT", -16),
             bundle: None,
         },
         HeightInfo {
             height: Height::Chocolate,
-            epoch: -17,
+            epoch: get_upgrade_height("FOREST_CHOCOLATE_HEIGHT", -17),
             bundle: None,
         },
         HeightInfo {
             height: Height::OhSnap,
-            epoch: -18,
+            epoch: get_upgrade_height("FOREST_OHSNAP_HEIGHT", -18),
             bundle: None,
         },
         HeightInfo {
             height: Height::Skyr,
-            epoch: -19,
+            epoch: get_upgrade_height("FOREST_SKYR_HEIGHT", -19),
             bundle: None,
         },
         HeightInfo {
             height: Height::Shark,
-            epoch: -20,
+            epoch: get_upgrade_height("FOREST_SHARK_HEIGHT", -20),
             bundle: Some(
                 Cid::try_from("bafy2bzacedozk3jh2j4nobqotkbofodq4chbrabioxbfrygpldgoxs3zwgggk")
                     .unwrap(),
             ),
         },
         HeightInfo {
             height: Height::Hygge,
-            epoch: -21,
+            epoch: get_upgrade_height("FOREST_HYGGE_HEIGHT", -21),
             bundle: Some(
                 Cid::try_from("bafy2bzacebzz376j5kizfck56366kdz5aut6ktqrvqbi3efa2d4l2o2m653ts")
                     .unwrap(),
             ),
         },
         HeightInfo {
             height: Height::Lightning,
-            epoch: -22,
+            epoch: get_upgrade_height("FOREST_LIGHTNING_HEIGHT", -22),
             bundle: Some(
                 Cid::try_from("bafy2bzaceay35go4xbjb45km6o46e5bib3bi46panhovcbedrynzwmm3drr4i")
                     .unwrap(),
             ),
         },
         HeightInfo {
             height: Height::Thunder,
-            epoch: -1,
+            epoch: get_upgrade_height("FOREST_THUNDER_HEIGHT", -1),
             bundle: None,
         },
         HeightInfo {
             height: Height::Watermelon,
-            epoch: 10,
+            epoch: get_upgrade_height("FOREST_WATERMELON_HEIGHT", 200),
```

Comment:
Perhaps it'd be more idiomatic to have it as `Result` so the usage would be `get_upgrade_height("FOREST_WATERMELON_HEIGHT").unwrap_or(200)`. Not a big deal, though.

Reply:
I feel `get_upgrade_height("FOREST_WATERMELON_HEIGHT").unwrap_or(200)` suggests `FOREST_WATERMELON_HEIGHT` is defined in normal cases, so I'd prefer keeping it this way, just like lotus.

```go
getUpgradeHeight := func(ev string, def abi.ChainEpoch) abi.ChainEpoch {
		hs, found := os.LookupEnv(ev)
		if found {
			h, err := strconv.Atoi(hs)
			if err != nil {
				log.Panicf("failed to parse %s env var", ev)
			}

			return abi.ChainEpoch(h)
		}

		return def
	}
```

Reply:
Lotus has clunky Go error handling and C-like notion of optional values. How about an `Option<ChainEpoch>`?

Reply:
As for the function name `get_upgrade_height` I'd prefer `get_upgrade_height(&str, i64) -> i64`, or I can rename it to `get_upgrade_height_from_env_var(&str) -> Option<i64>`, what do you think?

Reply:
IMO the 2nd one is more expressive and leaves less room for guessing. I'd omit the `_var` in the name.

Reply:
@LesnyRumcajs fixed